Add proposal for terminating KREAd vats

Jorge-Lopes opened the pull request:
closes: https://github.com/Agoric/BytePitchPartnerEng/issues/46

# Description

As part of the sunsetting process for the KREAd application, it is necessary to terminate the KREAd contract, governor, committee, and charter vats. This step is essential to ensure a clean and secure shutdown of the application and to free up associated resources.

This PR introduces:
- A new core-eval script to handle the termination of the KREAd-related vats.
- An a3p-integration proposal to validate the script's behavior.

## Key Changes
Core-Eval Script:
- Added a new script to execute the termination of KREAd vats.

A3P-Integration Enhancements:
- Copied the agoric-sdk/a3p-integration directory, including the z:acceptance proposal.
- Removed kread.test.js from the acceptance test phase.
- Introduced a new x:kread proposal to test the KREAd termination process.

Environment Setup:
- Added a .env file requiring the AGORIC_SDK_ROOT variable to be updated with the full path to the agoric-sdk root directory.
- Included a script (agoric-sdk-symlink.sh) to create a symbolic link to the agoric-sdk.


# Testing Considerations

The x:kread proposal was introduced to test the behaviour of the executing the KREAd terminating core-eval, as well as test how it handles new offer requests to the KREAd contract after terminated.
The z:acceptance proposal confirms that the Agoric chain remains functional after the termination of KREAd vats.

Note that it is necessary to update the value of `AGORIC_SDK_ROOT` at the agoric/a3p-integration/.env with the one from your local environment. And then execute the agoric/a3p-integration/agoric-sdk-symlink.sh script. 
After executing these 2 steps, the remaining steps are the same as usual.


# Additional Considerations

After merging this PR, the next logical step would be to create a new release. This release should include artifacts generated by the agoric run command, which can then be utilized with the Cosmos Proposal Builder.
Generated artifacts will be available in agoric/a3p-integration/proposals/x:kread/kread-submission.

Jorge-Lopes committed:
chore(proposal): add proposal for terminating relevant contracts

rel: https://github.com/Agoric/BytePitchPartnerEng/issues/46

Jorge-Lopes committed:
fix(proposal): update sourceSpec path

anilhelvaci commented:
please see https://github.com/Agoric/BytePitchPartnerEng/issues/46#issuecomment-2585709162

Chris-Hibbert commented:
> n:upgrade-next which allows testing against the current chain state.
The present state of `n: upgrade-next` reflects tests that upgrade-18 succeeds. Now that U18 has been run on mainNet [#211](https://github.com/Agoric/agoric-3-proposals/pull/211) will remove most of those tests and move a few to `z:acceptance`. This PR should probably mostly follow that.

`x:kread/test.sh` doesn't invoke ava. Doesn't it need `yarn ava ./*.test.js` or `yarn ava` or something in order to start the tests?

The test calls `evalBundles('kread-submission')`, but there's no such directory checked in, nor instructions in the `package.json` to generate it. What's the plan?

Jorge-Lopes commented:
> The present state of `n: upgrade-next` reflects tests that upgrade-18 succeeds. Now that U18 has been run on mainNet [#211](https://github.com/Agoric/agoric-3-proposals/pull/211) will remove most of those tests and move a few to `z:acceptance`. This PR should probably mostly follow that.

The purpose of including the `n:upgrade-next` proposal was to test the KREAd termination in a state that already included the [PR #10267](https://github.com/Agoric/agoric-sdk/pull/10267).
If I understand your suggestion correctly, since U18 is now on mainNet, there is no longer a need to include this proposal, since the agoric-3-proposals:latest image should provide the necessary chain state. 
@Chris-Hibbert Can you please confirm if this understanding is accurate?

> `x:kread/test.sh` doesn't invoke ava. Doesn't it need `yarn ava ./*.test.js` or `yarn ava` or something in order to start the tests?

The are no post-build tests for the `x:kread`proposal, only during the eval phase.
I’ve updated the test.sh script to make this clearer.

> The test calls `evalBundles('kread-submission')`, but there's no such directory checked in, nor instructions in the `package.json` to generate it. What's the plan?

Thank you very much for bringing this to my attention. I failed to notice that the submissions were not being commited since they are included in the gitignore. I’ve fixed this issue

Chris-Hibbert commented:
I'm confused. The new state only has one changed file (src/proposal/terminate-kread.js). I was expecting to see an updated `a3p-integration` test based on post-upgrade-18 chain state. This development may have to wait on the post-U18 cleanup.

> since U18 is now on mainNet, there is no longer a need to include this proposal, since the agoric-3-proposals:latest image should provide the necessary chain state.

After [#211](https://github.com/Agoric/agoric-3-proposals/pull/211), the ability to terminate vats should be present and `n:upgrade-next` should be unnecessary. But we'll still need to add `x:kread` in order to kill the various vats, and `z:acceptance` to observe that the chain returns to health afterwards.

Jorge-Lopes commented:
> I'm confused. The new state only has one changed file (src/proposal/terminate-kread.js). I was expecting to see an updated `a3p-integration` test based on post-upgrade-18 chain state. This development may have to wait on the post-U18 cleanup.

@Chris-Hibbert sorry, I have decided to improve my implementation to allow the use of the `yarn build` command without the previous limitations.
As you suggested, I removed the n:upgrade-next.

Jorge-Lopes committed:
chore(a3p): copy a3p-integration along with acceptance proposal

Jorge-Lopes committed:
chore(a3p): add x:kread proposal

Jorge-Lopes committed:
chore(a3p): add symbolic link script
